### PR TITLE
feat: If AppGroup defined in plist apply group dir.

### LINKED ISSUE #702: Support for App Groups in MMKV 3.0.0-beta.6
Hello,

I'm currently developing an iOS share extension and I believe that the MMKV library would be a great fit for my project. MMKV's compatibility with both React Native versions and its support for app extensions make it an ideal choice. However, I noticed that although the documentation mentions support for app groups via info.plist, this feature seems to be missing in release 3.0.0-beta.6.

As someone who is not an iOS developer, I have managed to implement the basic functionality for sharing app groups. However, having this feature natively supported in MMKV would significantly simplify development and improve the overall experience.

Could you please consider adding support for app groups in the upcoming releases? Thank you.

//  MmkvHostObject.cpp
```
#ifdef __APPLE__
#include <Foundation/Foundation.h>
  NSString* appGroup = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppGroup"];
    if( appGroup ){
        NSString* groupDir = [[NSFileManager defaultManager]
                              containerURLForSecurityApplicationGroupIdentifier:appGroup]
            .path;
            const char* groupDirCStr = [groupDir UTF8String];
            std::string groupDirStr(groupDirCStr);
            
            instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, &groupDirStr);

    }else{
        instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, pathPtr);
    }
```

### PR DESCRIPTION
**Pull request by calism2**
This commit adds functionality to check if an AppGroup is defined in the info.plist. If it is, the group directory is applied, allowing for shared storage between the main app and its extensions.

Closes #702

### CODE REVIEW

File: package/cpp/MmkvHostObject.cpp [hunk 1]
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
   MMKVMode mode = getMMKVMode(config);
 
 #ifdef __APPLE__
-  instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, pathPtr);
+#include <Foundation/Foundation.h>
```
Comment (mrousavy) [inline package/cpp/MmkvHostObject.cpp:31]: Please move the include up to the other header includes. Don't forget to wrap it in `#ifdef __APPLE__`  again.

File: package/cpp/MmkvHostObject.cpp [hunk 2]
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
...
+  NSString* appGroup = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"AppGroup"];
+  if (appGroup) {
```
Comment (mrousavy) [inline package/cpp/MmkvHostObject.cpp:33]:
```suggestion
  if (appGroup != nil) {
```

File: package/cpp/MmkvHostObject.cpp [hunk 3]
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
...
+    MmkvLogger::log("RNMMKV", "Path ignored, AppGroup defined plist");
```
Comment (mrousavy) [inline package/cpp/MmkvHostObject.cpp:34]:
```suggestion
    MmkvLogger::log("RNMMKV", "Info.plist specifies a custom AppGroup - ignoring `path`...");
```

File: package/cpp/MmkvHostObject.cpp [hunk 4]
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
...
+    NSString* groupDir =
+        [[NSFileManager defaultManager] containerURLForSecurityApplicationGroupIdentifier:appGroup]
+            .path;
+    const char* groupDirCStr = [groupDir UTF8String];
+    std::string groupDirStr(groupDirCStr);
```
Comment (mrousavy) [inline package/cpp/MmkvHostObject.cpp:39]:
```suggestion
    std::string groupDirStr = groupDir.UTF8String;
```

File: package/cpp/MmkvHostObject.cpp [hunk 5]
```diff
@@ -28,7 +28,21 @@ MmkvHostObject::MmkvHostObject(const facebook::react::MMKVConfig& config) {
...
+
+    instance = MMKV::mmkvWithID(config.id, mode, encryptionKeyPtr, &groupDirStr);
+
```
Comment (mrousavy) [inline package/cpp/MmkvHostObject.cpp:42]:
```suggestion
```

**Review by mrousavy — CHANGES_REQUESTED** [review]
thanks for the PR! Just a few minor nitpicks.

**Comment by calism2** [comment]
Hi,

Thank you for your fast response. I have implemented the requested changes. Additionally, if the AppGroup capability is not added correctly, the app will throw a null pointer exception, which might be difficult for JavaScript developers to trace. Therefore, I have added a custom runtime error to handle this scenario.

**Comment by mrousavy** [comment]
Hey! Thanks for the updates, but I felt like moving `getAppGroup` into `MmkvPlatformContext` made more sense, as it is iOS-specific code and I don't want to cram that into the shared .cpp file.

Let me know what you think: https://github.com/mrousavy/react-native-mmkv/pull/705

**Pull request closed without merging**